### ISSUE TITLE
reproduce the ppa

### ISSUE THREAD

**Issue by Juanhui28** [melifluos/subgraph-sketching#3]
Dear authors,

Thanks for providing the code. We have some issues to reproduce the results for the ppa dataset. I tuned the parameters by myself:`--lr `in  the range of `[0.01, 0.001]`, `--label_dropout `in the range of `[0.1, 0.3,0.5]`, `--feature_dropout` in the range of `[0.1, 0.3, 0.5]`, `--hidden_channels` to be `256`, and I also add the` --use_RA `to include the RA feature.  By selecting the parameter based on the validation performance, I got the hit100 result around 38 which is quite lower than 49 as reported in the paper.

To reproduce the result, what paramter do I need to tune? Could you please kindly give some suggestions?

**Comment by Juanhui28** [comment]
@melifluos @shir994 
If you know anything about the above issue, please let me know, thanks!

**Comment by melifluos** [comment]
lr looks a bit low, but that wouldn't explain the difference. I haven't tested the new code on PPA, so let me do that and get back to you. Also I think PPA is a slower to converge so needs a few more epochs, maybe 100.

**Comment by melifluos** [comment]
@Juanhui28 looking into this one now.

**Comment by Juanhui28** [comment]
Thanks!

**Comment by voladorlu** [comment]
@melifluos Thanks for sharing the code so much. I tried to run the code on PPA, but it fails at the subgraph feature extracting stage. It looks like taking too much memory close to 150GB. Is it because the number of edges are too large?

**Comment by melifluos** [comment]
Yes, that's correct, you need quite a lot of RAM for PPA and citation2

**Comment by voladorlu** [comment]
@melifluos Thanks for response. It probably needs to divide the edges into batches, then process them.

**Comment by melifluos** [comment]
> 

@voladorlu yeah, good idea. This has been added and will be merged shortly

**Comment by melifluos** [comment]
> Thanks!

@Juanhui28 PPA works better with these two flags that were not previously set in the README 
--use_feature 0 
--add_normed_features 1
They have now been added and I can recreate the results from the paper using the command in the README, but please let me know if you can't

**Comment by Juanhui28** [comment]
Sure, thank you so much!
